### PR TITLE
[#5861] improvement(CLI): Refactor the validation logic in the handle methods

### PR DESCRIPTION
**Pull request by Abyss-lord**
### What changes were proposed in this pull request?

refactor the validation logic of all entities and add test case, just like validation of table command #5906 . A hint is provided when the user's output is missing the required arguments. for example:

```bash
gcli column list -m demo_metalake, --name Hive_catalog
# Malformed entity name.
# Missing required argument(s): schema, table

gcli column details -m demo_metalake, --name Hive_catalog --audit
# Malformed entity name.
# Missing required argument(s): schema, table, column

gcli user delete -m demo_metalake
Missing --user option.
```
Currently, the Role command needs to be refactored and opened as a separate issue

### Why are the changes needed?

Fix: #5861 

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

local test

### CODE REVIEW

**Comment by Abyss-lord** [comment]
Hi @justinmclean @xunliu @tengqm , could you please review this PR when you have time? I’d really appreciate your feedback.

File: clients/cli/src/main/java/org/apache/gravitino/cli/ErrorMessages.java [hunk 1]
```diff
@@ -32,6 +32,7 @@ public class ErrorMessages {
   public static final String MISSING_NAME = "Missing --name option.";
   public static final String MISSING_GROUP = "Missing --group option.";
   public static final String MISSING_USER = "Missing --user option.";
+  public static final String MISSING_TAG = "Missing --tar option.";
```
Comment (tengqm) [inline clients/cli/src/main/java/org/apache/gravitino/cli/ErrorMessages.java:35]: typo

**Review by tengqm — COMMENTED** [review]
Regarding the scope, this PR looks good to me other than a typo.
However, I would NOT structure the validation of options/args this way. I'd delegate the validation to those command objects. I.e.

- Treat this class as a factory of command objects and focus
  on creating and invoking the correct commands.
- Let the command objects do their own validation and execution.
- To avoid a huge refactoring, we can treat parameter validation a part of the `handle()` workflow.

However, this belongs to a different (series of) PRs.

**Comment by justinmclean** [comment]
@tengqm see #5861 - this is currently waiting for review/merge.

**Comment by justinmclean** [comment]
Adding a validate method or similar to commands has already been proposed.

**Review by justinmclean — APPROVED** [review]
Thanks for all the work here

**Pull request merged**